alfresco-solrapi-client/alfresco-solrapi-client-ditto/src/main/java/eu/xenit/alfresco/solrapi/client/ditto/SolrModelMapper.java

```diff
@@ -0,0 +1,191 @@
+package eu.xenit.alfresco.solrapi.client.ditto;
+
+import eu.xenit.alfresco.client.solrapi.api.model.NodeNamePaths;
+import eu.xenit.alfresco.client.solrapi.api.model.NodePathInfo;
+import eu.xenit.alfresco.client.solrapi.api.model.SolrNodeMetaData;
+import eu.xenit.alfresco.client.solrapi.api.query.NodeMetaDataQueryParameters;
+import eu.xenit.alfresco.solrapi.client.ditto.dto.SolrNodeMetaDataModel;
+import eu.xenit.testing.ditto.api.data.ContentModel.Content;
+import eu.xenit.testing.ditto.api.model.ContentData;
+import eu.xenit.testing.ditto.api.model.MLText;
+import eu.xenit.testing.ditto.api.model.Node;
+import eu.xenit.testing.ditto.api.model.ParentChildAssoc;
+import eu.xenit.testing.ditto.api.model.QName;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class SolrModelMapper {
+
+    public SolrNodeMetaData toApiModel(SolrNodeMetaDataModel model) {
+        return new SolrNodeMetaData(
+                model.getId(),
+                model.getAclId(),
+                model.getTxnId(),
+                model.getNodeRef(),
+                model.getType(),
+                model.getProperties(),
+                model.getAspects(),
+                model.getPaths(),
+                model.getNamePaths(),
+                model.getAncestors(),
+                model.getParentAssocs(),
+                model.getParentAssocsCrc(),
+                model.getChildAssocs(),
+                model.getChildIds(),
+                model.getOwner(),
+                model.getTenantDomain());
+
+    }
+
+    public Function<Node, SolrNodeMetaDataModel> toSolrModel(NodeMetaDataQueryParameters params) {
```

Comment:
Can't this be a more straight forward `public SolrNodeMetaDataModel toSolrModel(Node node, NodeMetaDataQueryParameters params)` ?

Reply:
AS-IS usage: `.map(toSolrModel(params))`
With this change, it would be: `.map(n -> toSolrModel(node, params)`

Do we change it to the latter?

Reply:
Thinking more about it: this does not belong in the same "mapper" class as the method above!

As a generic concept you have:
api-model <---> client-model <---> dto-model
-spring: `SolrNodeMetaData` <---> `SolrNodeMetaDataModel` <---> json
-ditto: `SolrNodeMetaData` <---> `SolrNodeMetaDataModel` <---> `Node` (ditto)

Reply:
-ditto: SolrNodeMetaData <---> SolrNodeMetaDataModel <---> Node (ditto)

Why do we need this intermediate model here? 
For Spring it makes sense because the JSON response is deserialized into a `SolrNodeMetaDataModel`. However I don't really see the added value for this in the fake implementation?

Reply:
Because in this case, we'll indeed need to add two mappers: 
* `SolrNodeMetaDataModel` <---> Node (ditto)
* `SolrNodeMetaData`

However we can achieve what we want with a single one: 
* `SolrNOdeMetadata` <--> `Node`

?

Reply:
The first one contains logic that looks at include-filters etc.
The second one maps your internal model to what you want to offer in your API.

It's maybe a small thing now, but they are different things and are ideally decoupled.

Reply:
As per discussion:

* A shared internal model (`SolrNodeMetadataModel`) would be part of a clean architecture. 
* However right now, this shared model is just copied in the different subprojects. If we want to do this in a clean way, we should include another subproject including the **internal** model. 
* Right now, we choose to KISS and just remove the intermediate `SolrNodeMetaDataModel` from the fake implementation.

Reply:
Thanks for the summary @kerkhofsd